amlip_cpp/src/cpp/dds/multiservice/impl/AsyncMultiServiceClient.ipp

```diff
@@ -221,43 +237,36 @@ void AsyncMultiServiceClient<Data, Solution>::send_request_async_routine_()
 template <typename Data, typename Solution>
 void AsyncMultiServiceClient<Data, Solution>::read_replies_()
 {
+    // Read messages until the one expected (targeted to this id) is listen
     while (!should_stop_)
     {
-        // Only start waiting for data once a request have been sent
-        // This way it avoids to wait forever when no required
-        auto reason = waiter_for_request_.wait_and_decrement();
+        // TODO: this should not be done
+        uint32_t timeout = 1000;
+        auto reason = reply_available_reader_.wait_data_available(timeout);
         if (reason != eprosima::utils::event::AwakeReason::condition_met)
         {
-            // Waiter has been disabled
-            break;
+            continue;
         }
 
-        // Read messages until the one expected (targeted to this id) is listen
-        while (true)
+        // Get task reference
+        types::MsReferenceDataType reference = reply_available_reader_.read();
```

Comment:
Is it not necessary to wait for data availability?

Reply:
it waits in line 245. If it is awaken for other reason than data availability, it does not reach this code